mafiasi/gprot/views.py

```diff
@@ -455,3 +458,27 @@ def delete_reminder(request, reminder_pk):
     reminder.delete()
 
     return redirect('gprot_reminders')
+
+@login_required
+def favorite(request):
+    action = request.POST.get('action', None)
+    user = request.user
+    url = request.POST.get('url', None)
+    if action == 'save':
+        if Favorite.objects.filter(user=user, url=url):
+              return HttpResponse('already a favorite', status=400)
+        favorite = Favorite.objects.create(user=user, url=url)
+        keywords = re.split("search=", url)[1:]
+        for keyword in keywords:
+            if keyword[-1] == "&":
+                keyword = keyword[:-1]
+            if keyword[0] == "c":
+                favorite.courses.add(Course.objects.get(id=int(keyword[2:])))
+            elif keyword[0] == "e":
+                favorite.examiners.add(Teacher.objects.get(id=int(keyword[2:])))
```

Comment:
Check out `urllib.parse.parse_qs` to parse a query string. Maybe Django even has a function to parse its own GET queries, but I don't know.
Also, it might by better to use `get_object_or_404` for courses and teachers as well to avoid errors when invalid ids are passed.
After changing the `favorite` object, you should call `favorite.save()` to save the changes.